app/_src/mesh/guides/create-a-mesh-using-terraform-and-konnect.md

```diff
@@ -0,0 +1,602 @@
+}
+```
+
+You can override that variable by setting the environment variable `TF_VAR_k8s_cluster_config_path`.
+
+We need to create a system account and a token to authenticate the zone:
+
+```hcl
+resource "konnect_system_account" "zone_system_account" {
+  depends_on = [konnect_mesh_control_plane.my_meshcontrolplane]
+  name            = "mesh_${konnect_mesh_control_plane.my_meshcontrolplane.id}_${var.zone_name}"
+  description     = "Terraform generated system account for authentication zone ${var.zone_name} in ${konnect_mesh_control_plane.my_meshcontrolplane.id} control plane."
+  konnect_managed = false
+}
+
+resource "konnect_system_account_role" "zone_system_account_role" {
+  depends_on = [konnect_system_account.zone_system_account]
+  account_id       = konnect_system_account.zone_system_account.id
+  entity_id        = konnect_mesh_control_plane.my_meshcontrolplane.id
+  entity_region    = var.region
+  entity_type_name = "Mesh Control Planes"
+  role_name        = "Admin" # should be "Connector"
+}
+
+resource "time_offset" "one_year_from_now" {
+  offset_years = 1
+}
+
+resource "konnect_system_account_access_token" "zone_system_account_token" {
+  depends_on = [konnect_system_account.zone_system_account, time_rotating.rotate, time_offset.one_year_from_now]
+  account_id = konnect_system_account.zone_system_account.id
+  expires_at = time_offset.one_year_from_now.rfc3339
+  name       = konnect_system_account.zone_system_account.name
+}
+```
+
+We need to create a namespace and a secret to hold the Zone token:
+
+```hcl
+resource "kubernetes_namespace" "kong_mesh_system" {
+  metadata {
+    name = "kong-mesh-system"
+    labels = {
+      "kuma.io/system-namespace" = "true"
+    }
+  }
+}
+
+resource "kubernetes_secret" "mesh_cp_token" {
+  metadata {
+    name = "cp-token"
+    namespace = "kong-mesh-system"
+  }
+
+  data = {
+    token = konnect_system_account_access_token.zone_system_account_token.token
+  }
+
+  type = "opaque"
+
+  depends_on = [kubernetes_namespace.kong_mesh_system]
+}
+```
+
+And create a values file called `values.tftpl` with templated values for zone, address and control plane id:
+
+```yaml
+kuma:
+  controlPlane:
+    mode: zone
+    zone: ${zone_name}
+    kdsGlobalAddress: grpcs://${region}.mesh.sync.konghq.com:443
+    konnect:
+      cpId: ${cp_id}
+    secrets:
+      - Env: KMESH_MULTIZONE_ZONE_KDS_AUTH_CP_TOKEN_INLINE
+        Secret: cp-token
+        Key: token
+  ingress:
+    enabled: true
+  egress:
+    enabled: true
+```
+
+And finally we can create a zone deployment using `helm_release`:
+
+```hcl
+variable "zone_name" {
+  type        = string
+  description = "The name of the cluster."
+  default     = "zone1"
+}
+
+resource "helm_release" "kong_mesh" {
+  name       = "kong-mesh"
+  repository = "https://kong.github.io/kong-mesh-charts"
+  chart      = "kong-mesh"
+  version    = "{{ KM_VERSION }}"
+
+  namespace        = "kong-mesh-system"
```

Comment:
probably the same here `kubernetes_secret.kong_mesh_system.metadata.name`

Reply:
```suggestion
  namespace = kubernetes_namespace.kong_mesh_system.metadata.0.name
```